sdk/core/azure-core/inc/azure/core/http/websockets/websockets.hpp

```diff
@@ -0,0 +1,368 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Azure Core APIs implementing the WebSocket protocol [RFC 6455]
+ * (https://www.rfc-editor.org/rfc/rfc6455.html).
+ */
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/transport.hpp"
+#include "azure/core/internal/client_options.hpp"
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+  namespace _detail {
+    class WebSocketImplementation;
+  }
+  namespace _internal {
+
+    enum class WebSocketFrameType : int
```

Comment:
`WebSocketFrameType` vs `WebSocketFrameKind`, The guidelines should be consistent on how to name these.  We aren't consistent in the SDK and use both patterns.

Reply:
In a quick survey of the source code, it appears that the vast majority of the SDKs use "Type" as a suffix to distinguish types of things (KeyType, ContentType, CertificateContentType, BlockType). 

The only exception I could find was the SpanKind enumeration, but SpanKind is a term of art in OpenTelemetry.

So I think "XxxType" is what we should be using.